glide.yaml

```diff
@@ -109,8 +132,10 @@ import:
   - pkg/util/validation/field
   - pkg/util/wait
   - pkg/util/yaml
+  - pkg/version
   - pkg/watch
 - package: k8s.io/apiserver
+  version: kubernetes-1.9.2-beta.0
```

Comment:
Its amazing that we now are at a point where a mapping tag is available for staging repos!

Reply:
yes, most of the repos in kubernetes are tagged with release version.

Reply:
Release versions have always been available, but my understanding is that federation development needs to be able to chase master to ensure compatibility over a given release cycle.  Has that thinking changed?

Reply:
i did not get why federation development needs to chase kubernetes master. we should ideally be jumping from one stable version to another stable version.

Reply:
Stable release versions of federation should absolutely target stable release versions of kubernetes.  I'm not sure that would be ideal for master, though.  The risk of compatibility problems being introduced between federation and kubernetes during development of a release is greater if federation development only targets stable release versions vs updating a pin against master on a regular basis.

Reply:
Partially agree with your concern, but we need to consider that we are consumers of the kubernetes code and it would make sense for us to vendor in stable code as opposed to under development code. In worse cases if something breaks because of compatibility issue we can resort to bumping to the particular commit which can fix the issue. So, IMHO, the bumping strategy should be from one stable release to another stable release.
Also if you think practically, we didn't face any such issues in past couple of months, given that our federated clusters in ci jobs ran with latest master k8s code.

Reply:
There were many problems discovered by chasing trunk in previous releases, so I'm not sure a lack of recent evidence is justification for not doing so going forward.  The danger in not discovering problems as soon as possible is that the code introducing breakage will be 'baked in' and more difficult to back out or change.  The strategy for minimizing disruption to the federation development effort would be to only update to a new version of k/k and its dependencies once CI had validated its stability.  Only periodic jobs that validate against k/k HEAD would ever be broken, minimizing disruption for federation developers. 

That said, I'm not responsible for supporting the current codebase (v1) so do what you think is best.  For v2 (which should not depend on k/k), I will advocate strongly for keeping vendored staging repos as close to master as possible.